sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfig.java

```diff
@@ -68,12 +63,15 @@ public abstract class BigtableConfig implements Serializable {
   /** Weather validate that table exists before writing. */
   abstract boolean getValidate();
 
-  /** {@link BigtableService} used only for testing. */
-  abstract @Nullable BigtableService getBigtableService();
-
-  /** Bigtable emulator. Used only for testing. */
+  /** Bigtable emulator. */
   abstract @Nullable String getEmulatorHost();
 
+  /** User agent for this job. */
+  abstract @Nullable String getUserAgent();
+
+  /** Credentials for running the job. */
```

Comment:
Might want to mention that this will default the CredentialFactory in PipelineOptions/Gcpoptions if its null